RELEASE.md

```diff
@@ -0,0 +1,14 @@
+# Release Process
+
+1. Merge PR.
+2. Open a new PR to update CHANGELOG ([example](https://github.com/fastly/terraform-provider-fastly/pull/348)).
+3. Rebase latest remote master branch locally (`git pull --rebase origin master`).
+4. Tag a new release (`git tag -s vX.Y.Z -m "vX.Y.Z" && git push origin vX.Y.Z`).
+5. Copy/paste CHANGELOG into the [draft release](https://github.com/fastly/terraform-provider-fastly/releases).
+6. Publish draft release.
```

Comment:
I suggest we add a final step about communicating the new release via internal channels.

Reply:
Good shout 👍🏻